addons/account_asset/account_asset_invoice.py

```diff
@@ -46,6 +46,9 @@ def asset_create(self, cr, uid, lines, context=None):
         context = context or {}
         asset_obj = self.pool.get('account.asset.asset')
         for line in lines:
+            asset_id = asset_obj.search(cr, uid, [('code', '=', line.invoice_id.number)], context=context)
```

Comment:
asset_ids = asset_obj.search(cr, uid, [('code', 'in', lines.mapped('invoice_id').mapped('number'))], context=context)